File: Lua/07_oop_basics.lua
-- 07_oop_basics.lua
-- Lua uses metatables for OOP
local Person = {}
Person.__index = Person

function Person.new(name, age)
    local self = setmetatable({}, Person)
    self.name = name
    self.age = age
    return self
end

function Person:introduce()
    print("Hi, I'm " .. self.name .. " and I'm " .. self.age .. " years old.")
end

local p = Person.new("Emma", 27)
p:introduce()

File: Java/07_oop_basics.java
class Person {
    String name;
    int age;

    Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    void introduce() {
        System.out.println("Hi, I'm " + name + " and I'm " + age + " years old.");
    }
}

public class OopBasics {
    public static void main(String[] args) {
        Person p = new Person("Emma", 27);
        p.introduce();
    }
}
